spring-cloud-contract-tools/spring-cloud-contract-spec-pact/src/test/groovy/org/springframework/cloud/contract/verifier/spec/pact/PactContractConverterSpec.groovy

```diff
@@ -303,22 +330,245 @@ class PactContractConverterSpec extends Specification {
+						}
+					}
+				}
+			]
+		when:
+			Collection<Contract> contracts = converter.convertFrom(pactv2Json)
+		then:
+			contracts == expectedContracts
+	}
+
+	def "should convert from pact v3 to three SC contracts"() {
+		given:
+			Collection<Contract> expectedContracts = [
+				Contract.make {
+					description("java test interaction with a DSL array body")
+					request {
+						method(GET())
+						url("/")
+					}
+					response {
+						status(200)
+						headers {
+							contentType(applicationJson())
+						}
+						body([
+							[
+								"dob": "07/19/2016",
+								"id": 8958464620,
+								"name": "Rogger the Dogger",
+								"timestamp": "2016-07-19T12:14:39"
+							],
+							[
+								"dob": "07/19/2016",
+								"id": 4143398442,
+								"name": "Cat in the Hat",
+								"timestamp": "2016-07-19T12:14:39"
+							]
+					  	])
+						testMatchers {
+							jsonPath('$[0].id', byType())
+							jsonPath('$[1].id', byType())
+						}
+					}
+				},
+				Contract.make {
+					description("test interaction with a array body with templates")
+					request {
+						method(GET())
+						url("/")
+					}
+					response {
+						status(200)
+						headers {
+							contentType(applicationJsonUtf8())
+						}
+						body([
+							[
+								"dob": "2016-07-19",
+								"id": 1943791933,
+								"name": "ZSAICmTmiwgFFInuEuiK"
+							],
+							[
+								"dob": "2016-07-19",
+								"id": 1943791933,
+								"name": "ZSAICmTmiwgFFInuEuiK"
+							],
+							[
+									"dob": "2016-07-19",
+									"id": 1943791933,
+									"name": "ZSAICmTmiwgFFInuEuiK"
+							]
+					  	])
+						testMatchers {
+							jsonPath('$[2].name', byType())
+							jsonPath('$[0].id', byType())
+							jsonPath('$[1].id', byType())
+							jsonPath('$[2].id', byType())
+							jsonPath('$[1].name', byType())
+							jsonPath('$[0].name', byType())
+							jsonPath('$[0].dob', byDate())
+						}
+					}
+				},
+				Contract.make {
+					description("test interaction with an array like matcher")
+					request {
+						method(GET())
+						url("/")
+					}
+					response {
+						status(200)
+						headers {
+							contentType(applicationJsonUtf8())
+						}
+						body([
+						    "data": [
+								"array1": [[
+									"dob": "2016-07-19",
+									"id": 1600309982,
+									"name": "FVsWAGZTFGPLhWjLuBOd"
+								]],
+								"array2": [[
+									"address": "127.0.0.1",
+									"name": "jvxrzduZnwwxpFYrQnpd"
+								]],
+								"array3": [[
+									[
+										"itemCount": 652571349
+									]
+								]]
+							],
+							"id": 7183997828
+					  	])
+						testMatchers {
+							jsonPath('$.data.array3[0]', byType() {
+								maxOccurrence(5)
+							})
+							jsonPath('$.data.array1', byType() {
+								minOccurrence(0)
+							})
+							jsonPath('$.data.array2', byType() {
+								minOccurrence(1)
+							})
+							jsonPath('$.id', byType())
+
+//							TODO: CURRENTLY NOT SUPPORTED
```

Comment:
Why isn't it supported? That should work I guess OOB ?

Reply:
No, it wasn't. They added new matchers to v3, check https://github.com/pact-foundation/pact-specification/tree/version-3#add-an-equality-matcher until the generators